### PR TITLE
Add support for multiple issuers

### LINKED ISSUE #433: Allow multiple issuers?
Does it seem reasonable to allow PyJWT to accept a list of possible `iss`uer arguments, in the same way that `aud` does?  (It's possible I'm just lazy and trying to avoid try-catch blocks :joy:)

### PR DESCRIPTION
**Pull request by patrick91**
Closes #433

This PR adds support for specifing multiple issuers, similar to what we do for audience.

### CODE REVIEW

File: jwt/api_jwt.py [hunk 1]
```diff
@@ -212,7 +212,10 @@ def _validate_iss(self, payload, issuer):
         if "iss" not in payload:
             raise MissingRequiredClaimError("iss")
 
-        if payload["iss"] != issuer:
+        if isinstance(issuer, str):
```
Comment (jpadilla) [inline jwt/api_jwt.py:215]: do we also need to guard against other invalid values for `issuer`?

see https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py#L128-L129

**Review by jpadilla — CHANGES_REQUESTED** [review]

File: tests/test_api_jwt.py [hunk 2]
```diff
@@ -419,6 +419,12 @@ def test_check_issuer_when_valid(self, jwt):
         token = jwt.encode(payload, "secret")
         jwt.decode(token, "secret", issuer=issuer, algorithms=["HS256"])
 
+    def test_check_multiple_issuers(self, jwt):
+        issuer = "urn:foo"
+        payload = {"some": "payload", "iss": "urn:foo"}
+        token = jwt.encode(payload, "secret")
+        jwt.decode(token, "secret", issuer=[issuer, "urn:bar"], algorithms=["HS256"])
```
Comment (dimaqq) [inline tests/test_api_jwt.py:426]: The spec insists on issuers being HTTPS URLs:
https://openid.net/specs/openid-connect-core-1_0.html#Terminology

```suggestion
        jwt.decode(token, "secret", issuer=[issuer, "https://other.test"], algorithms=["HS256"])
```

Comment (dimaqq) [reply tests/test_api_jwt.py:426]: (arguably `pyjwt` could be used in other contexts, not just oidc)

**Comment by peterfarrell** [comment]
I'm interested in this feature.

**Comment by ChegeBryan** [comment]
> I'm interested in this feature.

That makes two of us.

**Comment by ChegeBryan** [comment]
@jpadilla is support for multiple issuers coming soon?

**Comment by jpadilla** [comment]
there were some unresolved comments on this pull request, I'd gladly review again another.

**Pull request closed without merging**